src/tools/clippy/clippy_utils/src/consts.rs

```diff
@@ -423,14 +422,14 @@ impl<'a, 'tcx> ConstEvalLateContext<'a, 'tcx> {
                 let result = self
                     .lcx
                     .tcx
-                    .const_eval_resolve(
+                    .const_eval_resolve_for_typeck(
```

Comment:
```suggestion
                    .const_eval_resolve(
```
these are not type system constants